From a6518553ea37a50f03143fbbad93350a342f4f94 Mon Sep 17 00:00:00 2001
From: miaogongzi <miaogongzi0227@gmail.com>
Date: Thu, 14 May 2026 13:38:31 +0800
Subject: [PATCH] Update Makefile

Signed-off-by: miaogongzi <miaogongzi0227@gmail.com>
---
 package/firmware/intel-microcode/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/package/firmware/intel-microcode/Makefile b/package/firmware/intel-microcode/Makefile
index 2808aa6810d899..b183f5c9d92d10 100644
--- a/package/firmware/intel-microcode/Makefile
+++ b/package/firmware/intel-microcode/Makefile
@@ -39,6 +39,7 @@ endef
 
 define Build/Compile
 	IUCODE_TOOL=$(STAGING_DIR)/../host/bin/iucode_tool $(MAKE) -C $(PKG_BUILD_DIR)
+	rm -rf $(PKG_BUILD_DIR)/intel-ucode-ipkg
 	mkdir -p $(PKG_BUILD_DIR)/intel-ucode-ipkg
 	$(STAGING_DIR)/../host/bin/iucode_tool -q \
 		--write-firmware=$(PKG_BUILD_DIR)/intel-ucode-ipkg $(PKG_BUILD_DIR)/$(MICROCODE).bin