From 048275abf3335fce66693d8441b969902426937b Mon Sep 17 00:00:00 2001
From: miaogongzi <miaogongzi0227@gmail.com>
Date: Thu, 12 Mar 2026 10:43:16 +0800
Subject: [PATCH] Update Makefile

Signed-off-by: miaogongzi <miaogongzi0227@gmail.com>
---
 package/firmware/intel-microcode/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package/firmware/intel-microcode/Makefile b/package/firmware/intel-microcode/Makefile
index 224c878dd646ea..447bb9d04af5f9 100644
--- a/package/firmware/intel-microcode/Makefile
+++ b/package/firmware/intel-microcode/Makefile
@@ -39,7 +39,8 @@ endef
 
 define Build/Compile
 	IUCODE_TOOL=$(STAGING_DIR)/../host/bin/iucode_tool $(MAKE) -C $(PKG_BUILD_DIR)
-	mkdir $(PKG_BUILD_DIR)/intel-ucode-ipkg
+	rm -rf $(PKG_BUILD_DIR)/intel-ucode-ipkg
+	mkdir -p $(PKG_BUILD_DIR)/intel-ucode-ipkg
 	$(STAGING_DIR)/../host/bin/iucode_tool -q \
 		--write-firmware=$(PKG_BUILD_DIR)/intel-ucode-ipkg $(PKG_BUILD_DIR)/$(MICROCODE).bin
 endef
