From 12b075479782a7e15265c9e1f3e0b746607d3aa0 Mon Sep 17 00:00:00 2001
From: mgz0227 <miaogongzi0227@gmail.com>
Date: Thu, 12 Mar 2026 18:40:44 +0800
Subject: [PATCH] Update Makefile

---
 net/adguardhome/Makefile | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/feeds/packages/net/adguardhome/Makefile b/feeds/packages/net/adguardhome/Makefile
index 8dc5537d8a0d76..455e9aa009f7a3 100644
--- a/feeds/packages/net/adguardhome/Makefile
+++ b/feeds/packages/net/adguardhome/Makefile
@@ -76,9 +76,6 @@ define Package/adguardhome/install
 	$(INSTALL_DIR) $(1)/etc/config
 	$(INSTALL_CONF) ./files/adguardhome.config $(1)/etc/config/adguardhome
 
-	$(INSTALL_DIR) $(1)/etc/init.d
-	$(INSTALL_BIN) ./files/adguardhome.init $(1)/etc/init.d/adguardhome
-
 	$(INSTALL_DIR) $(1)/etc/sysctl.d
 	$(INSTALL_CONF) ./files/adguardhome.sysctl $(1)/etc/sysctl.d/50-adguardhome.conf
 