From 646c01e13be993c0d8b12aeaf9dbb54a8bfc925e Mon Sep 17 00:00:00 2001
From: miaogongzi <miaogongzi0227@gmail.com>
Date: Thu, 2 Apr 2026 19:46:58 +0800
Subject: [PATCH] Update Makefile

Signed-off-by: miaogongzi <miaogongzi0227@gmail.com>
---
 lang/rust/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/feeds/packages/lang/rust/Makefile b/feeds/packages/lang/rust/Makefile
index f8c2305ce5643..1fff9f1d70d3e 100644
--- a/feeds/packages/lang/rust/Makefile
+++ b/feeds/packages/lang/rust/Makefile
@@ -91,6 +91,7 @@ define Host/Compile
 	TARGET_CFLAGS="$(TARGET_CFLAGS)" \
 	$(PYTHON) $(HOST_BUILD_DIR)/x.py \
 		--build-dir $(HOST_BUILD_DIR)/build \
+		--ci false \
 		dist build-manifest rustc rust-std cargo llvm-tools rust-src
 endef
