--- a/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
+++ b/package/feeds/luci/luci-mod-network/htdocs/luci-static/resources/view/network/dhcp.js
@@ -347,6 +366,10 @@ return view.extend({
 			_('Authoritative'),
 			_('This is the only DHCP server in the local network.'));
 
+		s.taboption('general', form.Flag, 'dns_redirect',
+			_('DNS Redirect'),
+			_('Force redirect all local DNS queries to DNSMasq, a.k.a. DNS Hijacking.'));
+
 		o = s.taboption('general', form.Value, 'local',
 			_('Resolve these locally'),
 			_('Never forward these matching domains or subdomains; resolve from DHCP or hosts files only.'));