--- a/target/linux/x86/base-files/etc/board.d/02_network
+++ b/target/linux/x86/base-files/etc/board.d/02_network
@@ -39,6 +39,9 @@ traverse-technologies-geos)
 xiaoma-m12)
 	ucidef_set_interfaces_lan_wan "eth1 eth2 eth3 eth4 eth5 eth6" "eth0"
 	;;
+*)
+	[ "$(ip address | grep ^[0-9] | awk -F: '{print $2}' | sed "s/ //g" | grep '^[e]' | grep -cvE "(@|\.)")" -gt 2 ] && ucidef_set_interfaces_lan_wan "$(ip address | grep ^[0-9] | awk -F: '{print $2}' | sed "s/ //g" | grep '^[e]' | grep -vE "(@|\.|eth1)" | tr "\n" " " | sed "s/ $//")" "eth1"
+	;;
 esac
 board_config_flush

